lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java

```diff
@@ -360,24 +365,51 @@ private Optional<JourneyResponse> checkForProfileMatch(
     }
 
     private JourneyResponse buildF2FNoMatchResponse(
-            boolean areGpg45VcsCorrelated, AuditEventUser auditEventUser) throws SqsException {
+            boolean areGpg45VcsCorrelated,
+            AuditEventUser auditEventUser,
+            ContraIndicators contraIndicators)
+            throws SqsException, MitigationRouteConfigNotFoundException, ConfigException,
+                    UnsupportedMitigationRouteException {
         LOGGER.info(LogHelper.buildLogMessage("F2F return - failed to match a profile."));
         sendAuditEvent(
                 !areGpg45VcsCorrelated
                         ? AuditEventTypes.IPV_F2F_CORRELATION_FAIL
                         : AuditEventTypes.IPV_F2F_PROFILE_NOT_MET_FAIL,
                 auditEventUser);
+        var mitigatedCI = ciMitUtilityService.hasMitigatedContraIndicator(contraIndicators);
+        if (mitigatedCI.isPresent()) {
+            var mitigationRoute = ciMitUtilityService.getMitigatedCiJourneyStep(mitigatedCI.get());
+            if (mitigationRoute.isPresent()) {
+                if (mitigationRoute.get().getJourney().equals(JOURNEY_ENHANCED_VERIFICATION_PATH)) {
+                    return JOURNEY_ENHANCED_VERIFICATION_F2F_FAIL;
+                }
+                throw new UnsupportedMitigationRouteException(
+                        String.format(
+                                "Unsupported mitigation route: %s",
+                                mitigationRoute.get().getJourney()));
```

Comment:
Looking at this I think it might be more readable if the logic was inverted so `if (!mitigationRoute.get()...) { throw new Unsupported....}`, and then returne the journey event after.